### PR TITLE
Pass initator internaldata into all stage models

### PR DESCRIPTION
**Pull request by brettwold**
Provide method to obtain flow initiation package name

### CODE REVIEW

File: flow-base-api/src/main/java/com/aevi/sdk/flow/stage/BaseStageModel.java [hunk 1]
```diff
@@ -70,9 +72,23 @@ protected BaseStageModel(@Nullable Activity activity) {
...
+    /**
+     * returns the packagename of the client application that initiated this flow in the first place
+     *
+     * @return A package name or "UNKNOWN" if not known for some reason
+     */
+    public String getFlowInitiatorPackage() {
+        return getInternalData(androidComponentDelegate.getSenderInternalData(), FLOW_INITIATOR);
+    }
+
+    protected String getInternalData(@Nullable InternalData senderInternalData, String flowStage) {
```
Comment (andreas-borglin) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/stage/BaseStageModel.java]: String flowStage?

File: flow-base-api/src/main/java/com/aevi/sdk/flow/stage/ServiceComponentDelegate.java [hunk 2]
```diff
@@ -122,6 +128,20 @@ private void completeMessageStream() {
...
 
+    @Nullable
+    public static String getActivityRequestJson(Activity activity) {
+        return activity.getIntent().getStringExtra(ACTIVITY_REQUEST_KEY);
+    }
+
+    @Nullable
+    public static InternalData getFlowInitiatorInternalData(Activity activity) {
+        String json = activity.getIntent().getStringExtra(EXTRAS_FLOW_INITIATOR);
+        if(json != null) {
```
Comment (andreas-borglin) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/stage/ServiceComponentDelegate.java]: Formatting

File: flow-base-api/src/main/java/com/aevi/sdk/flow/stage/ServiceComponentDelegate.java [hunk 3]
```diff
@@ -122,6 +128,20 @@ private void completeMessageStream() {
         return helper;
     }
 
+    @Nullable
+    public static String getActivityRequestJson(Activity activity) {
```
Comment (andreas-borglin) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/stage/ServiceComponentDelegate.java]: Not a major thing - but not sure these really belong as static methods in this class?

File: flow-base-api/src/main/java/com/aevi/sdk/flow/service/BaseApiService.java [hunk 4]
```diff
@@ -48,6 +48,10 @@ protected void attachBaseContext(Context base) {
         internalData.setSenderPackageName(getPackageName());
     }
 
+    protected String getInternalData(@Nullable InternalData senderInternalData, String flowStage) {
```
Comment (andreas-borglin) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/service/BaseApiService.java]: Can't see for sure - but is this called? It looks like the internal data is passed into the processRequest method?

Comment (brettwold) [reply flow-base-api/src/main/java/com/aevi/sdk/flow/service/BaseApiService.java]: Yeap its used in `ActivityProxyService` and `BasePaymentFlowService` to get the stageName.

Might come in handy later if we need for anything else

**Pull request merged**